env/src/lib.rs

```diff
@@ -139,12 +165,76 @@ use std::env;
 
 use log::{Log, LogLevel, LogLevelFilter, LogRecord, SetLoggerError, LogMetadata};
 
-struct Logger {
+pub struct Logger {
     directives: Vec<LogDirective>,
     filter: Option<Regex>,
+    format: Box<Fn(&LogRecord) -> String + Sync + Send>,
 }
 
 impl Logger {
+
+    pub fn new() -> Logger {
+        Logger {
+            directives: Vec::new(),
+            filter: None,
+            format: Box::new(|record: &LogRecord|
+                format!("{}:{}: {}",
+                    record.level(), record.location().module_path(), record.args()
+                )
+            ),
+        }
+    }
+
+    pub fn filter(mut self, module: Option<&str>, level: LogLevelFilter) -> Self {
+        self.directives.push(LogDirective {name: module.map(|s| s.to_string()), level: level});
+        self
+    }
+
+    pub fn format<F: 'static>(mut self, format: F) -> Self
+        where F: Fn(&LogRecord) -> String + Sync + Send
+    {
+        self.format = Box::new(format);
+        self
+    }
+
+    pub fn parse(mut self, filters: Option<String>) -> Self {
```

Comment:
I think this'll want to take a `&str` instead of `Option<String>` as there's no need to call this for `None` and it also doesn't need to take ownership of the string.